src/compute-client/src/plan/transform/relax_must_consolidate.rs

```diff
@@ -9,3 +9,47 @@
 
 //! Placeholder module for an [crate::plan::transform::Transform] that infers
 //! physical monotonicity.
+
+use std::marker::PhantomData;
+
+use mz_repr::GlobalId;
+
+use crate::plan::interpret::{PhysicallyMonotonic, SingleTimeMonotonic};
+use crate::plan::transform::{BottomUpTransform, TransformConfig};
+use crate::plan::Plan;
+
+/// A transformation that takes the result of single-time physical monotonicity
+/// analysis and refines, as appropriate, the setting of the `must_consolidate`
+/// flag in monotonic `Plan` nodes with forced consolidation.
+#[derive(Debug)]
+pub struct RelaxMustConsolidate<T = mz_repr::Timestamp> {
+    _phantom: PhantomData<T>,
+}
+
+impl<T> RelaxMustConsolidate<T> {
+    pub fn new() -> Self {
+        RelaxMustConsolidate {
+            _phantom: Default::default(),
+        }
+    }
+}
+
+impl<'a, T> BottomUpTransform<'a, GlobalId, T> for RelaxMustConsolidate<T> {
+    type Info = PhysicallyMonotonic;
+
+    type Interpreter = SingleTimeMonotonic<'a, T>;
+
+    fn name(&self) -> &'static str {
+        "must_consolidate relaxation"
+    }
+
+    fn interpreter(config: &'a TransformConfig<GlobalId>) -> Self::Interpreter {
```

Comment:
What is `TransformConfig<GlobalId>` meant to model here - all objects identified by a `GlobalId` that are known to be logically or physically monotonic?

Reply:
Yes, exactly. My intention was to model external information that was necessary to a `transform` call on `RelaxMustConsolidate`. In this case, it was the [set of physically monotonic `GlobalId`s for an object being built in this single-time dataflow](https://github.com/vmarcos/materialize/blob/7d6b6e1d20db976e92bead77baf8818150828f85/src/compute-client/src/plan/mod.rs#L1919-L1932).

I could represent this alternatively as part of the `RelaxMustConsolidate` struct and create a separate instance per call if we mean to reserve `TransformConfig` for only global configuration settings relevant to multiple `Transform`s?

Reply:
I followed the suggestion to create `TransformConfig` as a struct instead and project the relevant field when creating the interpreter for `RelaxMustConsolidate`.